pkg/sdn/api/types.go

```diff
@@ -16,11 +16,16 @@ type ClusterNetwork struct {
 	metav1.ObjectMeta
 
 	Network          string
+	ClusterDef       []ClusterNetworkEntry
```

Comment:
The "unversioned" ClusterNetwork type (the one in pkg/sdn/api/types.go, as opposed to pkg/sdn/api/v1/types.go) is an internal-only object, not part of the API, so it is not subject to backward-compatibility rules. So you could get rid of the old Network and HostSubnetLength fields here and only have the new array.

The one catch with this is that if the unversioned type and the v1 type aren't identical, then you'll need to add some hand-written conversion functions, but this isn't difficult. There are no examples in pkg/sdn/api but you can see examples in other API groups (eg, pkg/image/api/v1/conversion.go).